src/wp-includes/global-styles-and-settings.php

```diff
@@ -317,5 +363,6 @@ function wp_theme_has_theme_json() {
  */
 function wp_clean_theme_json_cache() {
 	wp_cache_delete( 'wp_get_global_stylesheet', 'theme_json' );
+	wp_cache_delete( 'wp_get_global_settings', 'theme_json' );
```

Comment:
The cache clearing here is not working as it is the wrong key: We need to clear the correct cache keys for all possible origins, i.e. `"wp_get_global_settings_{$origin}"`. What are the possible origins? We should probably loop through them here and clear those caches like that.

Reply:
Good catch, thanks! Fixed at https://github.com/WordPress/wordpress-develop/pull/3789/commits/8a47ba6c47c875a0062347f39ebc67f6c496b5cb